src/commands/exile_commands.py

```diff
@@ -93,3 +94,33 @@ async def roulette(interaction: discord.Interaction):
                     f"<@{interaction.user.id}> has tested their luck and lives another day...",
                     ephemeral=False,
                 )
+
+    @bot.tree.context_menu(name="Exile User for custom duration")
+    @discord.app_commands.check(is_user_moderator)
+    async def exile_user_action(interaction: discord.Interaction, user: discord.Member):
+        """Exile the selected user"""
+        await interaction.response.send_modal(ExileModal(user))
```

Comment:
will we hit the interaction timeout constraint if the DM hangs on us?

Reply:
The interaction doesn't seem to start until you press submit on the modal, so there's plenty of time to fill the form out. Possibly there's a timeout if you leave the modal up for the full discord timeout (I think it's 15 minutes?). And the DM just running a bit long is solved by having these all use the initial resp that we set up previously